src/plugins/pluginProxySharedModule_preBuild.ts

```diff
@@ -3,67 +3,106 @@ import { walk } from 'estree-walker';
             return {
               // Intercept all dependency requests to the proxy module
               // Dependency requests issued by localSharedImportMap are allowed without proxying.
-              find: new RegExp(`^${key}$`), replacement: preBuildLibPath, customResolver(source: string, importer: string) {
-                if (importer.includes(`node_modules/${key}/`)) {
-                  return (this as any).resolve(key)
-                }
-                config?.optimizeDeps?.needsInterop?.push(preBuildLibPath);
-                return (this as any).resolve(preBuildLibPath)
+              find: new RegExp(pattern), replacement: "$1", customResolver(source: string, importer: string) {
+                const loadSharePath = getLoadShareModulePath(source)
+                config?.optimizeDeps?.needsInterop?.push(loadSharePath);
+                // write proxyFile
+                writeLoadShareModule(source, shared[key], command)
+                writePreBuildLibPath(source)
+                addShare(source)
+                writeLocalSharedImportMap()
+                return (this as any).resolve(loadSharePath)
               }
             }
           })
         );
-        (config.resolve as any).alias.push(
-          ...Object.keys(shared).map((key) => {
-            return command === "build" ?
-              { find: new RegExp(`^${getPreBuildLibPath(key)}$`), replacement: key } :
-              {
-                find: new RegExp(`^${getPreBuildLibPath(key)}$`), customResolver(source: string, importer: string) {
-
-                  return (this as any).resolve(key)
+        const savePrebuild = new PromiseStore<string>()
+        
+          ; (config.resolve as any).alias.push(
+            ...Object.keys(shared).map((key) => {
+              return command === "build" ?
+                {
+                  find: new RegExp(`${virtualPackageName}/${PREBUILD_TAG}(.+)`), replacement: function (_: string, $1: string) {
+                    return packageNameDecode($1)
+                  }
+                } :
+                {
+                  find: new RegExp(`${virtualPackageName}/${PREBUILD_TAG}(.+)`), replacement: "$1", async customResolver(source: string, importer: string) {
+                    if (importer.includes(LOAD_SHARE_TAG)) {
+                      // save pre-bunding module id
+                      savePrebuild.set(source, (this as any).resolve(packageNameDecode(source)).then((item: any) => item.id))
+                    }
+                    // Fix localSharedImportMap import id
+                    return await (this as any).resolve(await savePrebuild.get(source))
+                  }
                 }
-              }
-          })
-        );
+            })
+          );
       },
     },
+    {
+      name: "watchLocalSharedImportMap",
+      apply: "serve",
+      config(config) {
+        if (!config.server) config.server = {}
```

Comment:
let's use [this](https://github.com/unjs/defu) for assiging default props here and elsewhere